curated_transformers/tokenization/tokenizer.py

```diff
@@ -162,6 +171,43 @@ def _convert_strings(
             for seq in input
         ]
 
+    def _validate_chunks(self, input: List[InputChunks]):
+        if self._any_special_tokens_regex is None:
+            special_tokens = self._special_tokens()
+            if len(special_tokens) == 0:
+                return
+            pattern = "|".join(
```

Comment:
I think we should use Aho-Corasick here. Much faster than non-regular language 'regular expressions` implementation and no special escaping needed:

https://pypi.org/project/pyahocorasick/